### DISCUSSION TITLE
Is RemafoX.swift required in an Objective-C project?

### DISCUSSION THREAD

**Discussion by cyphers** [FlineDev/RemafoX#107] [Help / FAQ] [answered]
Just getting started with RemafoX. All of our projects are strict Objective-C. So I threw the RemafoX.swift file in the trash assuming it was not relevant. I'm still muddling through and not entirely sure my assumptions are accurate as so much discussion here and in the videos is Swift-specific. Here are my questions.

1. Can I safely ignore the RemafoX.swift file as we do not use any Swift? We would like to continue to keep all Swift out of our projects.
2. How do I add a language? For testing, I started with English and Spanish. Now, I'm looking for the way to create a new copy of Localizable.strings in new languages with machine-translated entries in the new language. I previously was using auto_localize which asked for the languages I wanted and then generated the new strings files. It's not clear to me how this is done in RemafoX.
3. So far the lint and normalize features seem to be working. Machine translation worked on one file that was missing a few translations, so I know things are basically working, but adding languages seems to be the primary block.
4. In Xcode 15, they are pushing the new String Catalog files as the new way to do everything. It would be nice if RemafoX would support insertion of machine translations into String Catalog files. For now, it seems like I can't move forward to String Catalog files as long as I'm using RemafoX.

**Comment by Jeehut** [comment, marked as answer]
Hi @cyphers, thank you for taking the time to ask these questions. I'll try to answer them as good as I can:

> Can I safely ignore the RemafoX.swift file as we do not use any Swift? We would like to continue to keep all Swift out of our projects.

The `RemafoX.swift` file is a generated file which allows you to safely access your translations with full auto-completion support and failing builds if you removed a key from your translations and tried to reference it using the generated code. For example, a key "Onboarding.Page1.Title" can be called in Swift code via `Loc.Onboarding.Page1.Title.string` or (if you have turned on 'Generate Objective-C compatible code' in the 'Generated Code' configuration pane) in Objective-C via `LocCompat.onboarding_Page1_Title`.

If you prefer to continue using plain Strings without the provided additional safety, then you can safely ignore the `RemafoX.swift` file because at the end of the day, it's just a layer on top of Xcode's native `.strings(dict)` files. You will still profit from the Linting or Normalization features, for example.

> How do I add a language? For testing, I started with English and Spanish. Now, I'm looking for the way to create a new copy of Localizable.strings in new languages with machine-translated entries in the new language. I previously was using auto_localize which asked for the languages I wanted and then generated the new strings files. It's not clear to me how this is done in RemafoX.

RemafoX does not come with UI to add new languages to your project. It is not meant to replace any functionality that Xcode already provides. My goal is to fill in the gaps that Xcode leaves unfilled. To learn how to do this in Xcode, you can read this official documentation article, for example: https://developer.apple.com/documentation/xcode/adding-support-for-languages-and-regions

Tip: Before adding new languages, make sure to press the "Localize" button in the File Inspector on the right pane in Xcode for your `.strings(dict)` files. This way, Xcode will offer to add localized variants for you when adding a new language to the project.

RemafoX will automatically detect new added languages whenever the configuration file is opened for your project. You can also force-refresh the languages: If you've set up the Xcode extension like described in the app or in the linked videos, when in the 'Add Translation' window, simply press the 'Rescan Strings(dict) files' button at the top right.

By the way, if what you're looking for is to add a whole bunch of languages quickly, I've written a short guide explaining how to do this quickly in https://github.com/FlineDev/RemafoX/discussions/72.

> So far the lint and normalize features seem to be working. Machine translation worked on one file that was missing a few translations, so I know things are basically working, but adding languages seems to be the primary block.

Glad that everything else is working!

> In Xcode 15, they are pushing the new String Catalog files as the new way to do everything. It would be nice if RemafoX would support insertion of machine translations into String Catalog files. For now, it seems like I can't move forward to String Catalog files as long as I'm using RemafoX.

I'm already working on support for String Catalogs. I had a detailed look into them the first days after they were announced, and I also wrote this [FAQ article](https://www.fline.dev/the-missing-string-catalogs-faq-for-xcode-15/) for the community. I'm already done with the basic parser, but there's more work to do. I will do my best to get it ready before the final release of Xcode 15. After all, I want to migrate to String Catalogs myself. 😉
